src/test/TestDASHTree.cpp

```diff
@@ -141,89 +141,99 @@ TEST_F(DASHTreeTest, CalculateBaseUrlFromRedirect)
+  EXPECT_EQ(STR(segtpl->GetMediaUrl()), "https://foo.bar/media-video=66000-$Number$.m4s");
 }
 
 TEST_F(DASHTreeTest, CalculateSegTplWithBaseURLSlash)
 {
   // BaseURL inside period with trailing slash, uses segtpl, media/init doesn't start with slash
   OpenTestFile("mpd/segtpl_noslash_baseurl_slash.mpd", "https://foo.bar/initialpath/test.mpd");
 
-  adaptive::AdaptiveTree::SegmentTemplate segtpl =
-      tree->periods_[0]->adaptationSets_[1]->representations_[0]->segtpl_;
+  auto& segtpl = tree->m_periods[0]->GetAdaptationSets()[0]->GetRepresentations()[0]->GetSegmentTemplate();
 
-  EXPECT_EQ(segtpl.initialization, "https://foo.bar/guid.ism/dash/media-video=66000.dash");
-  EXPECT_EQ(segtpl.media_url, "https://foo.bar/guid.ism/dash/media-video=66000-$Number$.m4s");
+  EXPECT_EQ(STR(segtpl->GetInitialization()), "https://foo.bar/guid.ism/dash/media-video=66000.dash");
+  EXPECT_EQ(STR(segtpl->GetMediaUrl()), "https://foo.bar/guid.ism/dash/media-video=66000-$Number$.m4s");
 }
 
 TEST_F(DASHTreeTest, CalculateSegTplWithBaseURLAndMediaInitSlash)
 {
   // BaseURL inside period with trailing slash, uses segtpl, media/init starts with slash
   OpenTestFile("mpd/segtpl_slash_baseurl_slash.mpd", "https://foo.bar/initialpath/test.mpd");
 
-  adaptive::AdaptiveTree::SegmentTemplate segtpl =
-      tree->periods_[0]->adaptationSets_[1]->representations_[0]->segtpl_;
+  auto& segtpl = tree->m_periods[0]->GetAdaptationSets()[0]->GetRepresentations()[0]->GetSegmentTemplate();
 
-  EXPECT_EQ(segtpl.initialization, "https://foo.bar/media-video=66000.dash");
-  EXPECT_EQ(segtpl.media_url, "https://foo.bar/media-video=66000-$Number$.m4s");
+  EXPECT_EQ(STR(segtpl->GetInitialization()), "https://foo.bar/media-video=66000.dash");
+  EXPECT_EQ(STR(segtpl->GetMediaUrl()), "https://foo.bar/media-video=66000-$Number$.m4s");
 }
 
 TEST_F(DASHTreeTest, CalculateBaseURLInRepRangeBytes)
 {
   // Byteranged indexing
   OpenTestFile("mpd/segmentbase.mpd", "https://foo.bar/test.mpd");
-  EXPECT_EQ(tree->periods_[0]->adaptationSets_[1]->representations_[0]->url_,
+  EXPECT_EQ(tree->m_periods[0]->GetAdaptationSets()[0]->GetRepresentations()[0]->GetUrl(),
             "https://foo.bar/video/23.98p/r0/vid10.mp4");
 }
 
 TEST_F(DASHTreeTest, CalculateCorrectSegmentNumbersFromSegmentTimeline)
 {
+  tree->SetNowTime(1617223929L);
+
   // SegmentTimeline, availabilityStartTime is greater than epoch
   OpenTestFile("mpd/segtimeline_live_ast.mpd");
 
-  adaptive::SPINCACHE<adaptive::AdaptiveTree::Segment> segments =
-      tree->periods_[0]->adaptationSets_[0]->representations_[0]->segments_;
+  auto& segments =
+      tree->m_periods[0]->GetAdaptationSets()[0]->GetRepresentations()[0]->SegmentTimeline();
+
+  EXPECT_EQ(segments.GetSize(), 13);
+  EXPECT_EQ(segments.Get(0)->range_end_, 853663);
+  EXPECT_EQ(segments.Get(12)->range_end_, 853675);
+
+  /*! @todo: without SetNowTime this now return wrong range_end_ values, is this right?
```

Comment:
I think this might be wrong. For SegmentTimeline we definitely have to go by system clock but in this case it should be anchored to the start number.

Consider if my system clock is 2 minutes fast - the parser will have all of its segments in the future for this test manifest.

For easy reference, here is the template:
```
<SegmentTemplate initialization="$RepresentationID$/init.mp4?id=8c1bbb709229e110fbb0d1e544266504" 
  media="$RepresentationID$/segment_$Number$.m4s" 
  presentationTimeOffset="1738747575" 
  startNumber="487050"
  timescale="90000">
  <SegmentTimeline>
    <S d="540000" r="12" t="263007000000"/>
  </SegmentTimeline>
</SegmentTemplate>
```

Reply:
Really the segments need to always match the template since we have no idea if the duration and t values will be the same for future segments, if they will even exist.

Reply:
found the problem and fixed
now all tests pass ok